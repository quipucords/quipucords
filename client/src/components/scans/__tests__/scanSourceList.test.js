import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import ScanSourceList from '../scanSourceList';

describe('ScanSourceList Component', () => {
  const generateEmptyStore = () => configureMockStore()({});

  it('should shallow render a basic component with status', () => {
    const store = generateEmptyStore();
    const props = {
      scan: {
        sources: [
          {
            source_type: 'network'
          }
        ]
      }
    };
    const wrapper = shallow(<ScanSourceList {...props} />, { context: { store } });

    expect(wrapper.dive()).toMatchSnapshot();
  });
});
