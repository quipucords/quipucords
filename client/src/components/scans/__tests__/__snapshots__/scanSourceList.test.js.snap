// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanSourceList Component should shallow render a basic component with status 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={true}
>
  <Row
    bsClass="row"
    componentClass="div"
    key="15"
  >
    <Col
      bsClass="col"
      componentClass="div"
      md={3}
      xs={4}
    >
      <Icon
        name="network-range"
        type="pf"
      />
      Â  
      TestSource
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      md={9}
      xs={8}
    >
      Connection Scan: checking status...
    </Col>
  </Row>
</Grid>
`;
